lock.lua

```diff
@@ -0,0 +1,35 @@
+-- Simple class implementing mutex based on ngx.shared.dict:safe_add
```

Comment:
I did have a look at lua-resty-lock, but I decided not to use it because  it enforced dependency on resty.core. So far, this was not required in this module.

After reading the issue you linked I'm quite confident that it does not apply to this implementation of lock. The problem was in lua-resty-lock itself, because it's `__gc` function called the dict API at the same time it was already called. Nothing like that happens here.

Reply:
👍